libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,142 @@
+package protocols
+
+import (
+	"fmt"
+	"crypto/rand"
+	"encoding/hex"
+)
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const(
+	NONCE_BYTES_NB = 32
+)
+
+// type Dialogue struct {
+// 	_dialogues_storage
+// 	_dialogues_stats
+// 	_keep_terminal_state_dialogues
+// 	_message_class
+// 	_dialogue_class
+// 	_role_from_first_message
+// 	_self_address
+// 	_outgoing_messages
+// 	_incoming_messages
+// }
+
+// TODO
+// Define RoleType and figure out how to find role
+
+type Label struct {
+	dialogueReference [2]string
+	dialogueOpponentAddress Address
+	dialogueStarterAddress Address
+}
+
+type Dialogue struct {
```

Comment:
Missing: self_address, message_class which is associated with this dialogue, list of outgoing and incoming messages, list of ordered message ids, rules